File: OpenAlprWebhookProcessor/Settings/Agent.cs
using System;

namespace OpenAlprWebhookProcessor.Settings
{
    public class Agent
    {
        public string EndpointUrl { get; set; }

        public Guid Id { get; set; }

        public string Hostname { get; set; }

        public string Uid { get; set; }

        public string OpenAlprWebServerUrl { get; set; }

        public double? Latitude { get; set; }

        public double? Longitude { get; set; }

        public int SunriseOffset { get; set; }

        public int SunsetOffset { get; set; }

        public double TimezoneOffset { get; set; }

        public bool IsDebugEnabled { get; set; }

        public bool IsImageCompressionEnabled { get; set; }

        public long LastHeartbeatEpochMs { get; set; }

        public int? ScheduledScrapingIntervalMinutes { get; set; }

        public int? NextScrapeInMinutes { get; set; }
    }
}

File: OpenAlprWebhookProcessor/angularapp/src/app/settings/openalpr-agent/agent.ts
export class Agent {
    endpointUrl: string;
    id: string;
    isDebugEnabled: boolean;
    isImageCompressionEnabled: boolean;
    lastHeartbeatEpochMs: number;
    latitude: number;
    longitude: number;
    uid: string;
    openAlprWebServerUrl: string;
    nextScrapeInMinutes: Date;
    scheduledScrapingIntervalMinutes: number;
    sunriseOffset: number;
    sunsetOffset: number;
    timezoneOffset: number;

    constructor(init?: Partial<Agent>) {
        Object.assign(this, init);
    }
}
